docs/book/src/reference/using-finalizers.md

```diff
@@ -0,0 +1,99 @@
+# Using Finalizers
+
+`Finalizers` allow controllers to implement asynchronous pre-delete hooks. Let
+say, you create an external resource such as a storage bucket for each object of
+an API type you are implementing and you need to delete the associated external resource
+when the corresponding object is deleted from Kubernetes, you can use a
+finalizer to delete the external resource.
+
+You can read more about the finalizers in the [kubernetes reference docs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers). Section below
+demonstrates how to register and trigger pre-delete hooks in the `Reconcile`
+method of a controller.
+
+Highlights:
+- If object is not being deleted and does not have the finalizer registered,
+  then add the finalizer and update the object in kubernetes.
+- If object is being deleted and the finalizer is still present in finalizers list,
```

Comment:
we should probably note somewhere that a finalizer causes a "delete" to become an update to set deletion timestamps, so that's how you detect a delete.  Otherwise, without finalizers, a delete shows up as a reconcile where the object is missing from the cache.